arangod/Aql/types.h

```diff
@@ -60,10 +60,69 @@ struct Collection;
 /// @brief type for variable ids
 typedef uint32_t VariableId;
 
-/// @brief type for register numbers/ids
-typedef unsigned RegisterId;
 typedef size_t RegisterCount;
 
+/// @brief type for register numbers/ids
+struct RegisterId {
+  using value_t = unsigned;
+  RegisterId() = default;
+
+  static constexpr unsigned maxVariableRegisterId = 1000;
+  static constexpr unsigned constRegisterIdOffset = 2000;
```

Comment:
Can we use a way to distinguish regular and const registers that's easier to change later, if necessary? For example something like (just the idea)
```
struct Register {
    enum class Type { Regular, Const };
    Type _type : 1;
    unsigned _registerId : 10;
};
```
. That feels more closely related to a sum type (i.e. `type RegisterId = RegularRegister Int | ConstRegister Int`), and might be a little nicer to extend (e.g. for virtual registers).